community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -917,13 +1043,23 @@ public PrimitiveIntIterator relationshipGetPropertyKeys( KernelStatement state,
                 throw new EntityNotFoundException( EntityType.RELATIONSHIP, relationshipId );
             }
 
-            return Cursors.intIterator( relCursor.properties(), PropertyCursor.GET_KEY_INDEX_ID );
+            PrimitiveIntStack keys = new PrimitiveIntStack( 16 );
+            try (PropertyCursor properties = relCursor.properties())
+            {
+                while (properties.next())
+                {
+                    keys.push( properties.propertyKeyId() );
+                }
+            }
+
+            return keys.iterator();
```

Comment:
Wondering if it makes sense having a `EntityCursor` interface with the `properties()` method, so we can write this code once and use it for both node and relationships.